### PR TITLE
Target branch for PRs, call back function after pr is created

### PR DESCRIPTION
**Pull request by AgarFu**
### Description

Thanks for contributing this Pull Request. Make sure that you send in this Pull Request to the `master` branch of this repository, add a brief description, and tag the relevant issue(s) and PR(s) below.

### What Changed and Why? :
- Added CHANGELOG.md 
- Added an option to target different branches, not only master.
- Refactored `apply_transformation` function to allow injecting the list of repositories from different sources, not only the configuration file.
- Exposed a new function `transform`
```python
from inventory import list_of_repositories
transform(args, transformations, list_of_repositories)
```
- Added support for a callback function once the pull request is created, for example:  
```python
def dummy_callback(repo_name, pull_request):
    issue = new_issue(repo_name)
    pull_request.edit(title=pull_request.title + f' - [{issue.id}]')
```

- Type of change :
  - [x] New feature
  - [ ] Bug fix for existing feature
  - [x] Code quality improvement
  - [x] Addition or Improvement of tests
  - [ ] Addition or Improvement of documentation

- Other:
  - [x] Add unit tests
  - [ ] Add documentation

### CODE REVIEW

File: gordian/repo.py [hunk 1]
```diff
@@ -10,9 +10,9 @@
 BASE_URL = 'https://api.github.com'
 
 
-class Repo:
+class Repo(object):
```
Comment (jnevelson) [inline gordian/repo.py:13]: Why is this needed?

Comment (AgarFu) [reply gordian/repo.py:13]: Not needed it is there by default, something was not working and I was trying to change random things to "make it work", let me remove it.
